types/precison_timestamp_types.go

```diff
@@ -0,0 +1,167 @@
+package types
+
+import (
+	"fmt"
+	"github.com/cockroachdb/errors"
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+type TimePrecision int
+
+const (
+	UnknownPrecision TimePrecision = iota
+	Seconds
+	DeciSeconds
+	CentiSeconds
+	MilliSeconds
+	EMinus4Seconds // 10^-4 of seconds
+	EMinus5Seconds // 10^-5 of seconds
+	MicroSeconds
+	EMinus7Seconds // 10^-7 of seconds
+	EMinus8Seconds // 10^-8 of seconds
+	NanoSeconds
+)
```

Comment:
I'd prefer for `Precision` to be in the names of these. Otherwise `types.Seconds` could potentially be confusing as opposed to `types.SecondsPrecision` or something to that extent. 

Also, is there actually a need for the Deci/Centi/EMinus4/EMinus5, etc. types? Why is Seconds, Milliseconds, Microseconds, Nanoseconds not sufficient?

Reply:
> Why is Seconds, Milliseconds, Microseconds, Nanoseconds not sufficient?

The standard supports all sizes and so do most databases (0..9).

Reply:
I agree on Precision but I suggest we put first as opposed to last. Makes easier to find similar options in IDE. E.g. types.PrecisionMilliseconds

Reply:
fixed it, kindly recheck